### PR TITLE
add signingMessage to ConnectButton

### PR DESCRIPTION
**Pull request by akinolu52**
added the signingMessage props, so as to enable custom message to the <Connectbutton />

### CODE REVIEW

File: src/components/ConnectButton/ConnectButton.tsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ const {
 type web3StatusType = 'disconnected' | 'pending' | 'only_web3';
 
 const ConnectButton: React.FC<ConnectButtonProps> = ({
+    chainId = undefined,
```
Comment (BillyG83) [inline src/components/ConnectButton/ConnectButton.tsx:23]: Do you need to assign `undefined`?

I believe it is the same thing to have
```
chainId,
moralisAuth = true,
signingMessage,
```

**Review by BillyG83 — COMMENTED** [review]
looks great but one question for you on ConnectButton if you dont mind, TY @akinolu52

**Review by Y0moo — COMMENTED** [review]
The thing is that authentication happens in WalletModal and not in connect button.

**Comment by BillyG83** [comment]
TY @Y0moo did you get that feedback ok @akinolu52 ?

https://github.com/web3ui/web3uikit/pull/394#pullrequestreview-933319553

File: src/components/ConnectButton/ConnectButton.tsx (hunk 1 above)
Comment (akinolu52) [reply src/components/ConnectButton/ConnectButton.tsx:23]: it's removed (key and value) using `JSON.stringify`
if I dont use undefined i believed it'll be passed as an empty string to the backend, and i'm not sure if an empty string is being handled on the server. @BillyG83

**Comment by akinolu52** [comment]
Fixed in the latest update @Y0moo

**Review by Y0moo — APPROVED** [review]

File: src/components/WalletModal/types.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import { ConnectButtonProps } from '../ConnectButton/types';
 export interface WalletModalProps extends ConnectButtonProps {
+    /**
+     * an optional chain id of the blockchain that the web3 wallet is connected to
+     */
+    chainId?: number;
```
[the remark this answers is not in the document]
Comment (Y0moo) [reply src/components/WalletModal/types.ts:6]: checked, in this case, we need to sue number

**Pull request merged**